server/platform/services/sharedchannel/sync_send_remote.go

```diff
@@ -704,18 +765,26 @@ func (scs *Service) sendSyncMsgToRemote(msg *model.SyncMsg, rc *model.RemoteClus
 	err = rcs.SendMsg(ctx, rcMsg, rc, func(rcMsg model.RemoteClusterMsg, rc *model.RemoteCluster, rcResp *remotecluster.Response, errResp error) {
 		defer wg.Done()
 
-		var syncResp model.SyncResponse
-		if err2 := json.Unmarshal(rcResp.Payload, &syncResp); err2 != nil {
-			scs.server.Log().Log(mlog.LvlSharedChannelServiceError, "Invalid sync msg response from remote cluster",
-				mlog.String("remote", rc.Name),
-				mlog.String("channel_id", msg.ChannelId),
-				mlog.Err(err2),
-			)
+		// Check for ErrChannelNotShared in the response error
+		if errResp != nil && strings.Contains(errResp.Error(), ErrChannelNotShared.Error()) {
+			scs.handleChannelNotSharedError(msg, rc)
 			return
 		}
 
-		if f != nil {
-			f(syncResp, errResp)
+		var syncResp model.SyncResponse
+		if errResp == nil && rcResp != nil && len(rcResp.Payload) > 0 {
```

Comment:
We should either remove the `rcResp` checks from the condition or control and log that case in its `else` as an invalid sync response.

```suggestion
		if errResp == nil {
```

Reply:
re: "We should" - don't know if the extra checks hurt from a functional perspective. While unlikely to happen, if somehow `rcResp` is nil or has an empty payload despite no errResp error it is generally good practice to validate inputs before processing them, especially for data coming over a network. Favoring keeping it and logging the potential failure (your second option).

Reply:
made the change for option 2.